src/utils/source-search.js

```diff
@@ -48,26 +48,49 @@ function ignoreWhiteSpace(str) {
  * This returns a mode object used by CoeMirror's addOverlay function
  * to parse and style tokens in the file.
  * The mode object contains a tokenizer function (token) which takes
- * a character stream as input, advances it past a token, and returns
- * a style for that token. For more details see
+ * a character stream as input, advances it a character at a time,
+ * and returns style(s) for that token. For more details see
  * https://codemirror.net/doc/manual.html#modeapi
  *
+ * Also the token function code is mainly based of work done
+ * by the chrome devtools team. Thanks guys! :)
+ *
  * @memberof utils/source-search
  * @static
  */
 function searchOverlay(query) {
-  query = new RegExp(escapeRegExp(ignoreWhiteSpace(query)), "g");
+  query = new RegExp(escapeRegExp(ignoreWhiteSpace(query)));
+  let matchLength = null;
   return {
     token: function(stream) {
-      query.lastIndex = stream.pos;
-      let match = query.exec(stream.string);
-      if (match && match.index == stream.pos) {
-        stream.pos += match[0].length || 1;
-        return "selecting";
-      } else if (match) {
-        stream.pos = match.index;
-      } else {
-        stream.skipToEnd();
+      if (stream.column() === 0) {
+        matchLength = null;
+      }
+      if (matchLength !== null) {
```

Comment:
No, cos we want to run code from `line 82` if the matchLength is null because we might not have finished looking at all the characters in the file